src/api/system.c

```diff
@@ -377,13 +378,22 @@ static int f_poll_event(lua_State *L) {
         lua_pushstring(L, e.type == SDL_EVENT_WILL_ENTER_FOREGROUND ? "enteringforeground" : "enteredforeground");
         return 1;
       }
+
     case SDL_EVENT_WILL_ENTER_BACKGROUND:
       lua_pushstring(L, "enteringbackground");
       return 1;
+
     case SDL_EVENT_DID_ENTER_BACKGROUND:
       lua_pushstring(L, "enteredbackground");
       return 1;
 
+    case SDL_EVENT_WINDOW_DISPLAY_SCALE_CHANGED:
+    case SDL_EVENT_WINDOW_PIXEL_SIZE_CHANGED:
+      {
+        RenWindow* window_renderer = ren_find_window_from_id(e.window.windowID);
+        ren_resize_window(window_renderer);
+      }
```

Comment:
I wonder if we need `SDL_EVENT_WINDOW_DISPLAY_CHANGED` too.

Should we push the event to the lua side too? Or should we wait until we rewrite the scaling management?

Reply:
> SDL_EVENT_WINDOW_DISPLAY_CHANGED

IIRC this will help with changing the scale in Windows settings, I guess it wouldn't hurt to also react to the event.

> Should we push the event to the lua side too? Or should we wait until we rewrite the scaling management?

We can do this, but not much info I can push (I can call the function to get the scale), but right now its not that useful either.